backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/document/location/adapter/IDocumentShipFromLocationAdapter.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * #%L
+ * de.metas.adempiere.adempiere.base
+ * %%
+ * Copyright (C) 2021 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.document.location.adapter;
+
+import de.metas.bpartner.BPartnerContactId;
+import de.metas.bpartner.BPartnerId;
+import de.metas.bpartner.BPartnerLocationAndCaptureId;
+import de.metas.bpartner.BPartnerLocationId;
+import de.metas.bpartner.service.BPartnerInfo;
+import de.metas.document.location.DocumentLocation;
+import de.metas.document.location.RenderedAddressAndCapturedLocation;
+import de.metas.location.LocationId;
+import lombok.NonNull;
+import org.adempiere.exceptions.AdempiereException;
+
+import java.util.Optional;
+
+public interface IDocumentShipFromLocationAdapter extends IDocumentLocationAdapterTemplate
```

Comment:
why do we need IDocumentShipFromLocationAdapter?

what's the (logical) difference between this one and IDocumentDeliveryLocationAdapter?

Reply:
none, except different column names